python/tank/hook.py

```diff
@@ -206,6 +208,24 @@ def parent(self):
         """
         return self.__parent
 
+    def get_instance(self):
```

Comment:
Cool! I'll add a `create_hook_instance(hook_expression)` instead - that works nicely.

I did consider the whole "do it in the app" approach but the problem with that is that every hook needs to implement this, so it would make the hooks more complicated for users.